src/rt/lifetime.d

```diff
@@ -1193,24 +1186,53 @@ extern (C) void _d_delarray_t(void[]* p, const TypeInfo_Struct ti)
 {
     if (p)
```

Comment:
It's been there so far, but it seems gratitious. We might want to convert it to an assert.